File: filmforum_backend/src/main/java/rs/ac/bg/fon/filmforum_backend/dto/ReviewDTO.java
package rs.ac.bg.fon.filmforum_backend.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

import java.time.LocalDate;

public record ReviewDTO(
        long id,
        @NotBlank(message = "Review content can't be blank")
        String content,
        UserDTO author,
        LocalDate date,
        MovieDTO movie,
        @Min(value = 1, message = "Rating can't be less than 1")
        @Max(value = 10, message = "Rating can't be more than 10'")
        int rating

) {
}

File: filmforum_mobile/src/domain/ReviewDTO.ts
import {UserDTO} from './UserDTO';
import {MovieDTO} from './MovieDTO';

export interface ReviewDTO {
    id: number;
    content: string;
    author: UserDTO;
    date: string; // Assuming LocalDate is converted to string in the same format as ISO 8601 date format
    movie: MovieDTO;
    rating: number;
}
